mentii/mentii
# Activity mgmt

Adds the ability for a Teacher to add activities to the classes they are teaching.

Changes:
1. Added getBook endpoint.
2. Added getBookList endpoint, which returns a list of all bookId : bookTitle pairs.
3. Added getSampleProblems endpoint, which returns up to 4 sample problems, generated from the templates in the given book, chapter, and section.
4. Added addActivity endpoint which adds an activity to the given class, if the user is the teacher of the class.
5. Added book_ctrl.getBookList for getBookList endpoint.
6. Added class_ctrl.addActivity  for addActivity endpoint.
7. Modified ResponseCreation.py to allow for inclusion of decimals in json dump
8. Changed how Activities are displayed 
9. Added getBook, getAllBookTitlesAndIds getSampleProblems to book.service for above endpoints.
10. Added addActivity to class.service for above endpoint.
10. Created the Add Activity modal for the class details page.

How To Test:
1. Login as admin
2. Create a book with at least one chapter, section, and problem
3. Create a class, go to the details page for that class
4. Click the add new activity button
5. Confirm the add activity modal appears and functions as expected
6. Select your new book from the book drop down, and then the desired chapter and section
7. Confirm that 1-4 problems display from the section
8. Click the add new activity button on the modal
9. Confirm the new activity has been added to the class
10. repeat steps 4-9, but this time use the chapter 1 and section 1 of the Foundations of Algebra I text book
11. Try to mentor mentii from both of the activities you have created.
